### PR TITLE
fix discoveryAddress in configmap template

### PR DESCRIPTION
**Pull request by stevenctl**
As a result of making our integration tests have istiod-less remotes, we don't care what this is in the remote cluster, so this change wasn't caught by our tests.

This breaks remotes with an istiod in cluster, this causes the proxies to point to the local cluster for discovery, but it's only meant for injection + ca. 

should be a fix for #29389

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

**Comment by stevenctl** [comment]
Going to revert the integration tests so that we actually have verification on this.

**Review by linsun — APPROVED** [review]
Great catch - nice @stevenctl

**Comment by linsun** [comment]
Agreed this would fix #29389. I checked the istiod-remote endpoint or service etc, all look good.

cc @nmittler @irisdingbj

**Review by morvencao — APPROVED** [review]

**Comment by nmittler** [comment]
@stevenctl I'll approve once integ tests have been reverted to test this.

File: pkg/test/framework/components/istio/operator.go [hunk 1]
```diff
@@ -362,6 +362,7 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 	return i, nil
 }
 
+// patchIstiodCustomHost finds the Primary cluster for the given cluster, and
```
Comment (nmittler) [inline pkg/test/framework/components/istio/operator.go:365]: and ...?

File: pkg/test/framework/components/istio/flags.go [hunk 2]
```diff
@@ -50,4 +50,6 @@ func init() {
 		"Deploy Istio into the target Kubernetes environment with Helm.")
 	flag.BoolVar(&settingsFromCommandline.DumpKubernetesManifests, "istio.test.istio.dumpManifests", settingsFromCommandline.DumpKubernetesManifests,
 		"Dump generated Istio install manifests in the artifacts directory.")
+	flag.BoolVar(&settingsFromCommandline.IstiodlessRemotes, "istio.test.istio.istiodlessRemotes", settingsFromCommandline.IstiodlessRemotes,
+		"Dump generated Istio install manifests in the artifacts directory.")
```
Comment (nmittler) [inline pkg/test/framework/components/istio/flags.go:54]: Accidental copy+paste of the text from the variable above?

**Review by nmittler — COMMENTED** [review]
Looks good overall. Do we need another job to test istiod-less remotes? Or do we already have a job for external istiod (which implies istiod-less remotes)?

**Comment by stevenctl** [comment]
> Do we need another job to test istiod-less remotes? 
Yes, we should have 2 new jobs:
- One targeting external istiod running networking tests
- One targeting multicluster/multinetwork with istiodless remotes

> Or do we already have a job for external istiod (which implies istiod-less remotes)?

The externalistiod runs as a suite in the old multicluster job. It actually tests very little. We should get rid of it when the suggestion above is running.

**Pull request merged**